python/dolfinx/io/gmshio.py

```diff
@@ -330,7 +373,43 @@ def model_to_mesh(
     else:
         ft = meshtags(mesh, tdim - 1, np.empty(0, dtype=np.int32), np.empty(0, dtype=np.int32))
 
-    return (mesh, ct, ft)
+    if has_edge_data:
+        # Permute edges from MSH to DOLFINx ordering
+        edge_type = _cpp.mesh.cell_entity_type(
+            _cpp.mesh.to_type(str(ufl_domain.ufl_cell())), tdim - 2, 0
+        )
+        gmsh_edge_perm = cell_perm_array(edge_type, num_edge_nodes)
+        marked_edges = marked_edges[:, gmsh_edge_perm]
+
+        local_entities, local_values = distribute_entity_data(
+            mesh, tdim - 2, marked_edges, edge_values
+        )
+        mesh.topology.create_connectivity(topology.dim - 2, tdim)
+        adj = adjacencylist(local_entities)
+        et = meshtags_from_entities(mesh, tdim - 2, adj, local_values.astype(np.int32, copy=False))
+        et.name = "Edge tags"
+    else:
+        et = meshtags(mesh, tdim - 2, np.empty(0, dtype=np.int32), np.empty(0, dtype=np.int32))
+
+    if has_vertex_data:
+        # Permute vertices from MSH to DOLFINx ordering
+        vertex_type = _cpp.mesh.cell_entity_type(
+            _cpp.mesh.to_type(str(ufl_domain.ufl_cell())), tdim - 3, 0
+        )
+        gmsh_vertex_perm = cell_perm_array(vertex_type, num_vertex_nodes)
+        marked_vertices = marked_vertices[:, gmsh_vertex_perm]
+
+        local_entities, local_values = distribute_entity_data(
+            mesh, tdim - 3, marked_vertices, vertex_values
+        )
+        mesh.topology.create_connectivity(topology.dim - 3, tdim)
+        adj = adjacencylist(local_entities)
+        vt = meshtags_from_entities(mesh, tdim - 3, adj, local_values.astype(np.int32, copy=False))
+        vt.name = "Vertex tags"
+    else:
+        vt = meshtags(mesh, tdim - 3, np.empty(0, dtype=np.int32), np.empty(0, dtype=np.int32))
+
+    return GMSHModel(mesh, ct, ft, et, vt)
```

Comment:
GmshModel is consistent with our variable naming and Gmsh preferred capitalisation.

Reply:
Fixed in https://github.com/FEniCS/dolfinx/pull/3547/commits/bd5d73b916fc73687a0b214d4770dbef317e73b2